File: programmieren lernen/python/quelldateien/03_zahlenraten_kommentiert.py
# Das ist das Spiel "Zahlenraten" (oder "guess the number" in englisch):
# Das Programm denkt sich eine Zahl aus, die der Spieler möglichst schnell
# erraten muss

# das Modul "random" wird eingebettet,
# damit wir später eine Zufallszahl erzeugen können
import random

# Spieler begrüßen
print("Hallo! Wie heißt du?")
spielerName = input()

# Zahl zwischen 1 und 20 "ausdenken"
zahl = random.randint(1, 20)
print("Also, " + spielerName + ", ich habe mir eine Zahl zwischen 1 und 20 ausgedacht.")

# den Spieler höchstens 6 Runden raten lassen
runde = 0
while runde < 6:
    # wieder eine Runde mehr
    runde = runde + 1

    # Spieler eine Zahl raten lassen und in "geraten" merken
    print("Du darfst jetzt raten:")
    geraten = input()
    # von einer Zeichenkette in eine wirkliche Zahl umwandeln
    geraten = int(geraten)

    # wenn die geratene Zahl kleiner als die gedachte Zahl ist
    if geraten < zahl:
        print("Du hast zu niedrig geraten!")

    # wenn die geratene Zahl größer als die gedachte Zahl ist
    if geraten > zahl:
        print("Du hast zu hoch geraten")

    # wenn die geratene Zahl gleich der gedachten Zahl ist
    if geraten == zahl:
        # "runde" in eine Zeichenkette umwandeln, damit man sie ausgeben kann
        runde = str(runde)
        print("Gut gemacht, " + spielerName + ". Du hast die Zahl in " + runde + " Runden erraten!")
        # Spielschleife verlassen - wir müssen ja nicht mehr raten
        break

# wenn die geatene Zahl zum Schluss immer noch falsch ist
if geraten != zahl:
    # "zahl" in eine Zeichenkette umwandeln, damit man sie ausgeben kann
    zahl = str(zahl)
    print("Falsch. Ich habe an die " + zahl + " gedacht...")

File: programmieren lernen/cpp/quelldateien/03_zahlenraten_kommentiert.cpp
/**
Das ist das Spiel "Zahlenraten" (oder "guess the number" in Englisch)

Das Programm denkt sich eine Zahl aus, die der Spieler möglichst schnell erraten muss
**/

#include <iostream>
#include <stdlib.h>

using namespace std;

int main(void){ 


	int gesucht, eingabe, runde = 0;

	// Der Zufallszahlengenerator wird mit der Systemzeit initialisiert
	// Setzt also einen Startwert
	srand(time(NULL));

	// Zahl zwischen 1 und 20 "ausdenken"
	gesucht = rand() % 20 + 1;

	cout << endl << "Ich habe mir eine Zahl zwischen 1 und 20 ausgedacht.";

	do{
		// Eine Runde mehr
		runde++;

		// Den Spieler eine Zahl raten lassen ...
		cout << endl << "Du darfst jetzt raten: > ";
		// ... und die Zahl in "eingabe" speichern
		cin >> eingabe;

		// Wenn die Zahl kleiner als die gedachte Zahl ist:
		if(eingabe < gesucht){
			cout << endl << "Du hast zu niedrig geraten.";
		// Wenn die Zahl groesser als die gedachte Zahl ist:
		}else if(eingabe > gesucht){
			cout << endl << "Du hast zu hoch geraten.";
		// Wenn die Zahl gleich der gedachten Zahl ist:
		}else if(eingabe == gesucht){
			cout << endl << "Gut gemacht, du hast die Zahl in " << runde << " Runden erraten!";
			// Der Spieler hat gewonnen! Deshalb wird die Schleife abgebrochen
			break;
		}
	// Den Spieler höchstens 6 Runden raten lassen
	}while(runde < 6);

	// Wenn die Zahl am Schluss immernoch falsch ist:
	if(eingabe != gesucht){
		cout << endl << "Falsch. Ich habe an die " << gesucht << " gedacht...";
	}

	// Programm ordnungsgemäß beenden
	return 0;
}

